packages/eslint-plugin-internal/src/rules/prefer-ast-types-constant.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  AST_NODE_TYPES,
+  AST_TOKEN_TYPES,
+  ESLintUtils,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+
+const isStringLiteral = (
+  node: TSESTree.Literal,
+): node is TSESTree.StringLiteral => typeof node.value === 'string';
+
+export = ESLintUtils.RuleCreator(name => name)({
+  name: __filename,
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Best Practices',
+      recommended: 'error',
+      description:
+        'Ensures consistent usage of AST_NODE_TYPES & AST_TOKEN_TYPES.',
+    },
+    messages: {
+      preferConstant: 'Prefer {{ constant }}.{{ literal }} over raw literal',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const report = (
+      constant: 'AST_NODE_TYPES' | 'AST_TOKEN_TYPES',
+      literal: TSESTree.StringLiteral,
+    ): void =>
+      context.report({
+        data: { constant, literal: literal.value },
+        messageId: 'preferConstant',
+        node: literal,
+        fix: fixer =>
+          fixer.replaceText(literal, `${constant}.${literal.value}`),
+      });
+
+    return {
+      Literal(node: TSESTree.Literal): void {
+        if (!isStringLiteral(node)) {
+          return;
+        }
+
+        const value = node.value;
```

Comment:
one request, just because we have the actual definitions in this repo.
Could you do a parent check to not report on the enum definitions?

This will just save adding a disable check in the file.

```ts
if (
  node.parent?.type === AST_NODE_TYPES.TSEnumMember &&
  node.parent.parent?.type === AST_NODE_TYPES.TSEnumDeclaration &&
  ['AST_NODE_TYPES', 'AST_TOKEN_TYPES'].includes(node.parent.parent.id.name)
) {
  return;
}
```

Reply:
hahaha I didn't even consider that 🤦‍♂:joy: